search/controllers/api/__init__.py

```diff
@@ -108,8 +119,59 @@ def classic_query(params: MultiDict) -> Tuple[Dict[str, Any], int, Dict[str, Any
         HTTP status code.
     dict
         Extra headers for the response.
+
+    Raises
+    ------
+    :class:`BadRequest`
+        Raised when the search_query and id_list are not specified.
     """
-    pass
+    params = params.copy()
+    raw_query = params.get('search_query')
+
+    # parse id_list
+    id_list = params.get('id_list', '')
+    if id_list:
+        id_list = id_list.split(',')
+    else:
+        id_list = []
+
+    # error if neither search_query nor id_list are specified.
+    if not id_list and not raw_query:
+        raise BadRequest("Either a search_query or id_list must be specified for the classic API.")
+
+    if raw_query:
+        # migrate search_query -> query variable
+        params['query'] = raw_query
+        del params['search_query']
+
+        # pass to normal search, which will handle parsing
+        data, _, _ = search(params)
+    
+    if id_list and not raw_query:
+        # Process only id_lists.
+        # Note lack of error handling to implicitly propogate any errors. 
+        # Classic API also errors if even one ID is malformed.
```

Comment:
That's probably a feature. Better to complain loudly than proceed with unannounced errors